model/flow/epoch.go

```diff
@@ -239,20 +239,22 @@ func (part DKGParticipant) EncodeRLP(w io.Writer) error {
 // service events emitted as of the reference block. Events not yet emitted are
 // represented by ZeroID.
 type EpochStatus struct {
-	CurrentEpoch EventIDs // Epoch Preparation Events for the current Epoch
-	NextEpoch    EventIDs // Epoch Preparation Events for the next Epoch
+	FirstBlockID Identifier // ID of the first block in current epoch
```

Comment:
> If we are considering an EpochStatus to be invalid, if FirstBlockID is ZeroID, I would suggest to add an error return to the constructor:

👍 I'll move the `error` down to this layer like you suggest, so we guarantee at this layer that `EpochPhaseUndefined` only occurs along with an error. 

> However, from an external viewpoint, it seems like a totally fine value

👍  I'll update the documentation to be clear that this a zero/sentinel value as described in my other comment.

> not getting the Epoch Phase for the finalized block is handled like the happy path ; while in fact it should be treated as a fatal bug.

We do not treat errors from the protocol state that we expect to never happen as fatal errors practically anywhere in our codebase -- they cause the job/request being handled to stop and ultimately the error is logged. This is slightly different because we continue executing rather than return on the error, but I think generally the error handling is consistent. I don't agree that this component should decide to completely crash the node based on its assessment of the correctness of the protocol state. If we want to crash the node when we detect the protocol state is invalid, it would be much better to do that in the protocol state itself rather than at all the callsites into the protocol state.

Reply:
@jordanschalm to make an issue for panicking on fatal errors in protocol state

Reply:
https://github.com/dapperlabs/flow-go/issues/5076

Reply:
https://github.com/onflow/flow-go/pull/143/commits/85a9a9e96437e7d1845724c08fc37fa2eb231495

Reply:
nice cleanup thanks. 🎵